llvm/lib/Target/AMDGPU/R600InstrInfo.cpp

```diff
@@ -739,38 +740,36 @@ unsigned R600InstrInfo::insertBranch(MachineBasicBlock &MBB,
     if (Cond.empty()) {
       BuildMI(&MBB, DL, get(R600::JUMP)).addMBB(TBB);
       return 1;
-    } else {
-      MachineInstr *PredSet = findFirstPredicateSetterFrom(MBB, MBB.end());
-      assert(PredSet && "No previous predicate !");
-      addFlag(*PredSet, 0, MO_FLAG_PUSH);
-      PredSet->getOperand(2).setImm(Cond[1].getImm());
-
-      BuildMI(&MBB, DL, get(R600::JUMP_COND))
-             .addMBB(TBB)
-             .addReg(R600::PREDICATE_BIT, RegState::Kill);
-      MachineBasicBlock::iterator CfAlu = FindLastAluClause(MBB);
-      if (CfAlu == MBB.end())
-        return 1;
-      assert (CfAlu->getOpcode() == R600::CF_ALU);
-      CfAlu->setDesc(get(R600::CF_ALU_PUSH_BEFORE));
-      return 1;
     }
-  } else {
     MachineInstr *PredSet = findFirstPredicateSetterFrom(MBB, MBB.end());
     assert(PredSet && "No previous predicate !");
     addFlag(*PredSet, 0, MO_FLAG_PUSH);
     PredSet->getOperand(2).setImm(Cond[1].getImm());
+
     BuildMI(&MBB, DL, get(R600::JUMP_COND))
-            .addMBB(TBB)
-            .addReg(R600::PREDICATE_BIT, RegState::Kill);
-    BuildMI(&MBB, DL, get(R600::JUMP)).addMBB(FBB);
+        .addMBB(TBB)
+        .addReg(R600::PREDICATE_BIT, RegState::Kill);
     MachineBasicBlock::iterator CfAlu = FindLastAluClause(MBB);
     if (CfAlu == MBB.end())
-      return 2;
+      return 1;
```

Comment:
This is a functional change? And the same below.

Reply:
I think it is NFC, but git's diff is matching up the wrong lines. This is the "inner" version that is nested under the "then" part of `if (!FBB)`, but it is being compared against the "else" version from the original.

Reply:
I see, thanks!

Reply:
Yeah, you can get a more readable diff for that part if you select "Hide whitespace" under the gear wheel icon at the top of the whole diff.